kernel-cert-script

```diff
@@ -0,0 +1,86 @@
+#!/bin/sh
+
+op=$(echo "$0" | sed -e "s/.*-//")
+
+while true ; do
+    case $1 in
+	--kmp)
+	    kmp="$2"
+	    shift 2
+	    ;;
+	--certs)
+	    certs="$2"
+	    shift 2
+	    ;;
+	*) break
+	    ;;
+    esac
+done
+
+run_mokutil () {
+    [ -z "$KERNEL_PACKAGE_SCRIPT_DEBUG" ] || echo mokutil "$@" > /dev/kmsg
+    mokutil "$@"
+}
+
+[ -z "$KERNEL_PACKAGE_SCRIPT_DEBUG" ] || echo "cert $op" kmp: "$kmp" certs: "$certs" > /dev/kmsg
+
+mok_rc=0
+
+case $op in
+    pre)
+	;;
+    preun)
+	if command -v mokutil >/dev/null; then
+	    for cert in "$certs"; do
+		cert="/etc/uefi/certs/${cert}.crt"
+		ln "$cert" "$cert.delete" || mok_rc=$?
+	    done
+	fi
+	;;
+    post)
+	if command -v mokutil >/dev/null; then
+	    # Only apply CA check on the kernel package certs (bsc#1173115)
+	    if [ -z "$kmp" ] && mokutil -h | grep -q "ca-check"; then
+		MOK_ARG="--ca-check"
+	    else
+		MOK_ARG=""
+	    fi
+	    # XXX: Only call mokutil if UEFI and shim are used
+	    for cert in "$certs"; do
+		cert="/etc/uefi/certs/${cert}.crt"
+		run_mokutil --import "$cert" --root-pw ${MOK_ARG}
+		rc=$?
+		if [ $rc != 0 ] ; then
+		    mok_rc=$rc
+		    echo "Failed to import $cert"
+		fi
+	    done
+	fi
+	;;
+    postun)
+	if command -v mokutil >/dev/null; then
+	    for cert in "$certs"; do
+		cert="/etc/uefi/certs/${cert}.crt"
+		if ! test -e "$cert"; then
+		    run_mokutil --delete "$cert.delete" --root-pw
+		    rc=$?
+		    if [ $rc != 0 ] ; then
+			mok_rc=$rc
+			echo "Failed to delete $cert"
+		    fi
+		fi
+		rm "$cert.delete" || mok_rc=$?
+	    done
+	fi
+	;;
+    posttrans)
```

Comment:
Rather than looking at `$certs` in `%postun`, we might as well iterate over all `*.delete` files in `%posttrans`.

Reply:
Yes, that's also fine. It does not really save us anything, though.

The certificate is de-enrolled and then the flag file is deleted so it is not de-enrolled more than once, and de-enrolling is cheap anyway, it just moves the flag to a different place where it is read at boot time by some EFI application.